pyiceberg/table/__init__.py

```diff
@@ -1150,6 +1149,7 @@ def upsert(
         Returns:
             An UpsertResult class (contains details of rows updated and inserted)
         """
+        from pyiceberg.io.pyarrow import expression_to_pyarrow
```

Comment:
Should we add this one here as well:
```suggestion
        try:
            import pyarrow as pa
        except ModuleNotFoundError as e:
            raise ModuleNotFoundError("For writes PyArrow needs to be installed") from e

        from pyiceberg.io.pyarrow import expression_to_pyarrow
```
```

```